File: .tmp_py4013/batch_09_dump/3001_minimum_moves_to_capture_the_queen.js
// LeetCode 3001 - Minimum Moves to Capture The Queen
// https://leetcode.com/problems/minimum-moves-to-capture-the-queen/

var minMovesToCaptureTheQueen = function(a, b, c, d, e, f) {
    if (a === e && (c !== a || (d - b) * (d - f) > 0)) return 1;
    if (b === f && (d !== b || (c - a) * (c - e) > 0)) return 1;
    if (c - e === d - f && (a - e !== b - f || (a - c) * (a - e) > 0)) return 1;
    if (c - e === f - d && (a - e !== f - b || (a - c) * (a - e) > 0)) return 1;
    return 2;
};

File: .tmp_rs4013/rust_check13/3001_minimum_moves_to_capture_the_queen.rs
#![allow(dead_code, unused_variables, unused_mut, unused_assignments, unused_imports)]
struct Solution;
// LeetCode 3001 - Minimum Moves to Capture The Queen
// https://leetcode.com/problems/minimum-moves-to-capture-the-queen/

impl Solution {
    pub fn min_moves_to_capture_the_queen(a: i32, b: i32, c: i32, d: i32, e: i32, f: i32) -> i32 {
        if a == e && (c != a || (d - b) * (d - f) > 0) {
            return 1;
        }
        if b == f && (d != b || (c - a) * (c - e) > 0) {
            return 1;
        }
        if c - e == d - f && (a - e != b - f || (a - c) * (a - e) > 0) {
            return 1;
        }
        if c - e == f - d && (a - e != f - b || (a - c) * (a - e) > 0) {
            return 1;
        }
        2
    }
}
